### PR TITLE
CA1804 Deletion fix

### PR DESCRIPTION
**Pull request by Rolstenhouse**

### CODE REVIEW

**Comment by kschecht** [comment]
:+1:

**Comment by lgolding** [comment]
What is the Code Analysis warning that's being fixed here?

File: src/Sarif/Converters/ClangAnalyzerConverter.cs [hunk 1]
```diff
@@ -43,7 +43,6 @@ public void Convert(Stream input, IResultLogWriter output)
 
                 using (XmlReader xmlReader = XmlReader.Create(input, settings))
                 {
-                    XmlNodeType nodeType = xmlReader.MoveToContent();
```
Comment (lgolding) [inline src/Sarif/Converters/ClangAnalyzerConverter.cs]: I'm concerned about this one. I understand that the return value from `MoveToContent` -- the variable `nodeType` -- is not used. But presumably the call to `MoveToContent` has side effects (changing the internal state of the reader). Do the tests pass when you remove this call?

File: src/Sarif/Converters/ClangAnalyzerConverter.cs [hunk 2]
```diff
@@ -355,7 +354,7 @@ private void ReadPlistDictionary(XmlReader xmlReader, IList<Result> results)
                                     throw new InvalidDataException("Expected key value before dictionary data.");
                                 }
 
-                                string value = xmlReader.ReadElementContentAsString();
```
Comment (lgolding) [inline src/Sarif/Converters/ClangAnalyzerConverter.cs]: See, here's an example where you removed the variable but kept the call, which makes sense.

File: src/Sarif/Converters/FortifyPathElement.cs [hunk 3]
```diff
@@ -66,7 +66,6 @@ public static FortifyPathElement Parse(XmlReader xmlReader, FortifyStrings strin
                 throw xmlReader.CreateException(SdkResources.FortifyNotValidPathElement);
             }
 
-            int pathElementDepth = xmlReader.Depth;
```
Comment (lgolding) [inline src/Sarif/Converters/FortifyPathElement.cs]: And this one obviously has no side effects, so it's fine.

**Comment by lgolding** [comment]
This looks good except for the very first change. See my comment on that one.

**Comment by lgolding** [comment]
:+1:

**Comment by Rolstenhouse** [comment]
Added next commit before closing pull request, please check it out

File: src/Sarif/ExtensionMethods.cs [hunk 4]
```diff
@@ -153,15 +153,15 @@ public static void Populate(this Region region, NewLineIndex newLineIndex)
             }
         }
 
-        public static string GetMessageText(this Result result, IRule rule, bool concise = false)
```
Comment (lgolding) [inline src/Sarif/ExtensionMethods.cs]: The purpose is to allow the _caller_ to pass in `consise = true;`. I'll come see you about this one.

File: src/Sarif/ExtensionMethods.cs [hunk 5]
```diff
@@ -153,15 +153,19 @@ public static void Populate(this Region region, NewLineIndex newLineIndex)
             }
         }
 
+        public static string GetMessageText(this Result result, IRule rule)
+        {
+            return GetMessageText(result, rule, concise: false);
+        }
+
         public static string GetMessageText(this Result result, IRule rule, bool concise = false)
```
Comment (lgolding) [inline src/Sarif/ExtensionMethods.cs]: You added the overload which is good, but now you have to remove the default parameter:

```
public static string GetMessageText(this Result result, IRule rule, bool concise)
{
    ...
```

**Comment by lgolding** [comment]
:+1:

File: src/Sarif/ExtensionMethods.cs [hunk 6]
```diff
@@ -185,7 +185,19 @@ public static string GetMessageText(this Result result, IRule rule, bool concise
             {
                 throw new ArgumentNullException(nameof(rule));
             }
+            return GetMessageText(result, rule, concise: false);
```
Comment (lgolding) [inline src/Sarif/ExtensionMethods.cs]: A subtlety: When you have an overload that is just a _pass-through_ to an overload with more arguments, do all the argument checking in the overload with the most arguments. That way, you only have to write the argument-checking code once. So remove this check... unless... do you get another warning if you remove this check?

Also, formatting nitpick: our convention is to add a blank line after a closing brace. Likewise after line 200.

**Pull request merged**